src/build123d/topology.py

```diff
@@ -382,24 +356,31 @@ def tuplify(obj: Any, dim: int) -> tuple:
 class Mixin1D:
```

Comment:
```suggestion
class Mixin1D(Shape):
```
would avoid all the `assert isinstance(self, Shape)` in member methods

Reply:
I was trying to avoid doing this as the Mixin classes are defined before the Shape class in the topology.py file; however, I was forced to do this for other typing reasons and have removed all of the Shape assertions.  So ... much ... change.  There will be nothing left of the original file when I'm done :(